drivers/mge-hid.c

```diff
@@ -1151,6 +1151,119 @@ static info_lkp_t eaton_input_bypass_mode_off_info[] = {
+		bypass_switch_str = dstate_getinfo("input.bypass.switch.on");
+		if (!strcmp(bypass_switch_str, "disabled")) {
+			bypass_switch_str = eaton_input_bypass_check_range(value);
+		} else {
+			upsdebugx(1, "%s: Bypass switch on state is: %s , must be disabled before switching on", __func__, bypass_switch_str);
+			return NULL;
+		}
+
+		/* Check if input.eco.switchable is normal and set it to 'ECO' */
+		eco_switchable_str = dstate_getinfo("input.eco.switchable");
+		if (!strcmp(eco_switchable_str, "normal")) {
+			eco_switchable_str = eaton_input_eco_mode_check_range(value);
+		} else {
+			upsdebugx(1, "%s: ECO switch state is: %s , must be normal before switching to ECO", __func__, eco_switchable_str);
+			return NULL;
+		}
+
+		upsdebugx(1, "%s: ECO Mode was enabled after switching to Bypass Mode", __func__);
+        return "on";
+
+	} else {
+		/* Check if input.bypass.switch.off is disabled and set it to 'off' */
+		bypass_switch_str = dstate_getinfo("input.bypass.switch.off");
+		if (!strcmp(bypass_switch_str, "disabled")) {
+			setvar("input.bypass.switch.off", "off");
+		} else {
+			upsdebugx(1, "%s: Bypass switch off state is: %s , must be disabled before switching off", __func__, bypass_switch_str);
+			return NULL;
+		}
+
+		/* Check if input.eco.switchable is 'ECO' and set it to normal */
+		eco_switchable_str = dstate_getinfo("input.eco.switchable");
+		if (!strcmp(eco_switchable_str, "ECO")) {
+			setvar("input.eco.switchable", "normal");
+			buzzmode_set("vendor:mge-hid:normal");
+			/* Get the updated value of input.eco.switchable after setting it to "normal */
+            eco_switchable_str = dstate_getinfo("input.eco.switchable");
+		} else {
+			upsdebugx(1, "%s: ECO switch state is: %s , must be ECO before switching to normal", __func__, eco_switchable_str);
+			return NULL;
+		}
+
+		upsdebugx(1, "%s: ECO Mode was disabled after switching from Bypass Mode", __func__);
+    	return "off";
+	}
+}
+
+/* Conversion back of eaton_input_eco_mode_auto_on_off_fun() */
+static double eaton_input_eco_mode_auto_on_off_nuf(const char *value)
+{
+	const char *bypass_switch_str = NULL;
+	const char *eco_switchable_str = NULL;
+
+	if (!strcmp(value, "1")) {
+		bypass_switch_str = dstate_getinfo("input.bypass.switch.on");
+		if (!strcmp(bypass_switch_str, "disabled")) {
+			bypass_switch_str = eaton_input_bypass_check_range(1);
+		} else {
+			upsdebugx(1, "%s: Bypass switch on state is: %s , must be disabled before switching on", __func__, bypass_switch_str);
+			return 0.0;
+		}
+
+		eco_switchable_str = dstate_getinfo("input.eco.switchable");
+		if (!strcmp(eco_switchable_str, "normal")) {
+			eco_switchable_str = eaton_input_eco_mode_check_range(1);
+		} else {
+			upsdebugx(1, "%s: ECO switch state is: %s , must be normal before switching to ECO", __func__, eco_switchable_str);
+			return 0.0;
+		}
+
+		upsdebugx(1, "%s: ECO Mode was enabled after switching to Bypass Mode", __func__);
+		return 1.0;
+
+	} else {
```

Comment:
@masterwishx : I think this method should accept string `"0"` (or string `"off"`) as input here explicitly, and error out verbosely for unsupported inputs.